### DISCUSSION TITLE
how use extract_text and extract_table together.

### DISCUSSION THREAD

**Discussion by 88arvin** [jsvine/pdfplumber#891] [Ask for help with specific PDFs]
Can anyone help with this?
I have a pdf with multiple pages, and the structure of each page is the same, but the content of each page is different.
Each page consists of text and a table.
I want to extract the table and save it into a Pandas dataframe and I want to extract some text, save it in a new column(s), and then concatenate it into the Pandas dataframe.
Like this I want to iterate over the pages and extend the dataframe.

**Comment by 88arvin** [comment]
[pdf.pdf](https://github.com/jsvine/pdfplumber/files/11597057/pdf.pdf)
I want 'Period' as a column name and date as a value in the date column.

Output dataframe looks like:-
```
Period          Date     Narration     Chq/Ref No     Withdrawal (Dr)     Deposit(Cr)     Balance
01-04-2018               B/F                                                               0.00(Cr)
21-04-2022               B/F                                                               0.00(Cr)
```

**Reply by jsvine** [reply]
Hi @88arvin, this sounds more like a Python/Pandas question than a `pdfplumber` question, but I'll see if I can help. Is this what you're looking for?:

```python
subframes = []
for page in pdf.pages:
  text = my_function_to_extract_text(page)
  table = my_function_to_extract_table(page)
  combined = pd.DataFrame(table).assign(text=text)
  subframes.append(subframes)
df = pd.concat(subframes)
````
